babel-plugin-optimize-obj-str/readme.md

```diff
@@ -0,0 +1,214 @@
+# babel-plugin-optimize-obj-str
+
+> [Babel](https://babeljs.io/) plugin to optimize [`obj-str`](../) calls by replacing them with an equivalent unrolled expression.
+
+Even though the `obj-str` function is negligible in size over-the-wire, the motivation for this plugin is that transformed expressions [execute almost twice as fast as equivalent calls.](#performance)
+
+```js
+import objstr from 'obj-str';
+objstr({
+  'my-classname': true,
+  'another-one': maybe,
+  'third': a && b,
+});
+
+// Transformed:
+'' +
+  (true ? 'my-classname' : '') +
+  (maybe ? ' ' + 'another-one' : '') +
+  (a && b ? ' ' + 'third' : '');
+```
+
+## Install
+
+```
+npm install --save-dev babel-plugin-optimize-obj-str
+```
+
+## Usage
+
+**Via babel.config.js** ([recommended](https://babeljs.io/docs/en/configuration)):
+
+```js
+module.exports = {
+  plugins: ['optimize-obj-str'],
+};
+```
+
+**Via CLI**:
+
+```
+babel --plugins optimize-obj-str script.js
+```
+
+## Options
+
+```js
+module.exports = {
+  plugins: [
+    [
+      'optimize-obj-str',
+      {
+        strict: false,
+        referencesImport: ['obj-str', 'default'],
```

Comment:
I'm not sure if I like this option name, though I can't really offer alternatives. What about just `references` or `sources`?

Also, I assume AMP is doing something special with this, otherwise I'm not 100% sure what this option is for since the plugin is meant for `obj-str`

Reply:
I named it after the Babel method [`referencesImport()`](https://github.com/babel/babel/blob/d98418efbe6f1b4c7ad29a033b3de97f69a5e202/packages/babel-traverse/src/path/introspection.ts#L166). I've found that the names for equivalent options in other plugins vary wildly and I like none of them 😬

Reply:
Ok, so part two -- do we need this option at all?

Reply:
Not strictly. Removed.

Reply:
Thank you